components/brave_sync/brave_sync_prefs.cc

```diff
@@ -112,6 +114,8 @@ std::string Prefs::GetSeed() const {
   std::string seed;
   if (!OSCrypt::DecryptString(encrypted_seed, &seed)) {
     LOG(ERROR) << "Decrypt sync seed failure";
+    if (failed_to_decrypt)
+      *failed_to_decrypt = true;
     return std::string();
   }
   return seed;
```

Comment:
Since this is the only `return` with a valid string, then perhaps we should default `failed_to_decrypt` to yes at the top of the function and then set it to false here.

That way, if there's ever another check added to this function, the default will still be good.

Reply:
Yes, thanks, I will change that.

The normal case when `failed_to_decrypt` must be `false` but string result is expected to be empty - it is when  `pref_service_->GetString(kSyncV2Seed)` gave the empty string, which means there are no currently a sync chain and we will create new seed.

Reply:
@fmarier , this was changed in  https://github.com/brave/brave-core/pull/11935/commits/9204d4a5c261e44183640fefd84a11c21d2c3de1